### ISSUE TITLE
Fetch additional locals if needed.

### ISSUE THREAD

**Issue by rozzzly** [iamakulov/moment-locales-webpack-plugin#2]
It would be cool if the loader had an option (just a `boolean` flag) that would inject a little runtime into the bundle that checks if the locale is present in the whitelist. If it's not, it would fetch an additional bundle containing the locale.

Not sure how this would work exactly so far as what would happen during the time that the main bundle loaded, but the fetch for the additional locale(s) is still in progress; does it just act like `moment` normally does and give some "locale missing" error,  but once the fetch is complete give you the desired results? Or perhaps the module's evaluation is blocked until the locale is available (probably a terrible idea) or maybe an additional wrapper package for `moment` is created that exposes a promise which resolves when the desired locale is available—either imediately if the client's locale is in the whitelist, or after the fetch is complete.

Just a thought..

**Comment by iamakulov** [comment]
Interesting idea! Let me think about it.

Do you have any specific issue this would help you to solve?

**Comment by rozzzly** [comment]
No, not really. I just figure 98% percent of us are just targeting one/a few locale(s), which is why this loader is awesome. At the same time, it'd be cool if there was a gracefull fallback option for when that one random Korean guy visits your site.

**Comment by rozzzly** [comment]
No, not really. I just figure 98% percent of us are just targeting one/a few locale(s), which is why this loader is awesome. At the same time, it'd be cool if there was a gracefull fallback option for when that one random Korean guy visits your site.

**Comment by rozzzly** [comment]
No, not really. I just figure 98% percent of us are just targeting one/a few locale(s), which is why this loader is awesome. At the same time, it'd be cool if there was a gracefull fallback option for when that one random Korean guy visits your site.

**Comment by iamakulov** [comment]
Thought a bit, and I think this is out of scope for the plugin. Thanks for the suggestion, anyway :–)

Would be glad to re-open this though if there’s additional demand.
